crates/sui-core/src/node_sync/node_state.rs

```diff
@@ -562,21 +563,31 @@ where
             })
             .tap_err(|e| error!(?digest, "error: {}", e))?;
 
-        self.process_parents(permit, &digests.transaction, &effects)
-            .await?;
+        self.process_parents(
+            Some(permit),
+            &digests.transaction,
+            &effects,
+            PARENT_WAIT_TIMEOUT,
+        )
+        .await?;
 
         self.state
-            .handle_node_sync_certificate(cert, effects.clone())
+            .handle_node_sync_certificate(&cert, &effects)
             .await?;
 
         Ok(SyncStatus::CertExecuted)
     }
 
-    async fn process_parents(
+    /// Transaction initialized by Node Sync passes in a Semaphore for
```

Comment:
I see Node Sync, Node sync, node sync, and NodeSync all in comments in different files.

Reply:
what do yo recommend to use `Node Sync`?

Reply:
if not a proper noun or product name, you could use node sync generically. If it is a proper noun, use Node Sync